src/main/java/org/jabref/gui/externalfiles/FindFullTextAction.java

```diff
@@ -68,47 +88,61 @@ public void execute() {
 
             if (!confirmDownload) {
                 basePanel.output(Localization.lang("Operation canceled."));
-                return null;
+                return;
             }
         }
 
-        Map<Optional<URL>, BibEntry> downloads = new ConcurrentHashMap<>();
+        //Lookup full texts
+        BackgroundTask.wrap(this::findFullTexts)
+                .thenRun(this::downloadFullTexts)
+                .onFinished(this::displayDialogs)
```

Comment:
the onSuccess should already be on the FX thread

Reply:
Only one (usually the last) dialog message appears when downloading multiple full texts. What do you think the cause could be? Also, to clarify my commit message, the reference to the dialogs being broadcasted to the wrong thread was regarding the findFullTexts() method. Thanks for the help!